matminer/featurizers/structure.py

```diff
@@ -1550,3 +1574,297 @@ def citations(self):
+
+    def __init__(self, shells=(1, 2, 3), weight='area'):
+        """Initialize the featurizer
+
+        Args:
+            shells ([int]) - Which neighbor shells to evaluate
+            weight (str) - Attribute used to weigh neighbor contributions
+            """
+        self.shells = shells
+        self.weight = weight
+
+    def featurize(self, strc):
+        # Shortcut: Return 0 if there is only 1 type of atom
+        if len(strc.composition) == 1:
+            return [0]*len(self.shells)
+
+        # Get a list of types
+        elems, fracs = zip(*strc.composition.element_composition.fractional_composition.items())
+
+        # Precompute the list of NNs in the structure
+        voro = VoronoiNN(weight=self.weight)
+        all_nn = voro.get_all_nn_info(strc)
+
+        # Evaluate each shell
+        output = []
+        for shell in self.shells:
+            # Initialize an array to store the ordering parameters
+            ordering = np.zeros((len(strc), len(elems)))
+
+            # Get the ordering of each type of each atom
+            for site_idx in range(len(strc)):
+                nns = voro._get_nn_shell_info(all_nn, site_idx, shell)
+
+                # Sum up the weights
+                total_weight = sum(x['weight'] for x in nns)
+
+                # Get weight by type
+                for nn in nns:
+                    site_elem = nn['site'].specie.element \
+                        if isinstance(nn['site'].specie, Specie) else \
+                        nn['site'].specie
+                    elem_idx = elems.index(site_elem)
+                    ordering[site_idx, elem_idx] += nn['weight']
+
+                # Compute the ordering parameter
+                ordering[site_idx, :] = 1 - ordering[site_idx, :] / \
+                                            total_weight / np.array(fracs)
+
+            # Compute the average ordering for the entire structure
+            output.append(np.abs(ordering).mean())
+
+        return output
+
+    def feature_labels(self):
+        return ["mean ordering parameter shell {}".format(n) for n in self.shells]
+
+    def citations(self):
+        return ["@article{Ward2017,"
+                "author = {Ward, Logan and Liu, Ruoqian "
+                "and Krishna, Amar and Hegde, Vinay I. "
+                "and Agrawal, Ankit and Choudhary, Alok "
+                "and Wolverton, Chris},"
+                "doi = {10.1103/PhysRevB.96.024104},"
+                "journal = {Physical Review B},"
+                "pages = {024104},"
+                "title = {{Including crystal structure attributes "
+                "in machine learning models of formation energies "
+                "via Voronoi tessellations}},"
+                "url = {http://link.aps.org/doi/10.1103/PhysRevB.96.014107},"
+                "volume = {96},year = {2017}}"]
+
+    def implementors(self):
+        return ['Logan Ward']
+
+
+class StructureComposition(BaseFeaturizer):
```

Comment:
I think this should be deleted - the user can just create a composition column rather than pass through this wrapper object.

See for example the ``structure_to_composition`` function in ``conversions.py`` which allows for:

``df["composition"] = structure_to_composition(df["structure"])``

It is a *little* repetitive in the data since the composition is embedded in the structure, but I don't think it's so bad.

Reply:
Maybe Im being too hasty in suggesting deletion - there is some discoverability value in having this here. Lets discuss more

Reply:
I was concerned about this featurizer being a redundant with `structure_to_composition`, so I'm glad you brought this up.

My rationale for adding it was to allow for training a structure-based model with a single `MultiFeaturizer`. This way, I can write scripts that compare different structure-based featurizers easier - I do not need to have extra code for featurizers that use composition-based information and those which do not. 

The other advantage is that this allows for running featurizer on single entries (or non-pandas lists of entries) much more easily. Without this adapter, I might end up with code like this:

```python
x = Structure(...)
features = np.hstack(strc_f.featurize(s), comp_f.featurize(x.composition))
```

Which I think is tedious to write.

Do these use cases seem reasonable enough to warrant this wrapper class?
```

Reply:
Ok let's keep it